client/src/components/Query/Filter/index.jsx

```diff
@@ -1,6 +1,7 @@
 /* eslint-disable */
 
 import React from 'react';
+import { FormattedMessage } from 'react-intl';
```

Comment:
Would prefer passing `intl` as a prop, so to not enforce dependency over react-intl; we might want to manage language in a different fashion later on; such as using an api, but using a custom `intl.translate` function.

Reply:
Passing `intl` as a props still couples your components to `react-intl` since `intl` is imported from `react-intl` and your using a higher order component to pass it a props from a parent component.

 If you're ever curious about why you should prefer component, please refer to react-intl docs: https://github.com/formatjs/react-intl/blob/master/docs/Components.md#why-components

**Why Components?**
Beyond providing an idiomatic-React way of integrating internationalization into a React app, and the <Formatted*> components have benefits over always using the imperative API directly:

- Render React elements that seamlessly compose with other React components.
- Support rich-text string/message formatting in <FormattedMessage>.
- Implement advanced features like <FormattedRelativeTime>'s updating over time.
- Provide TypeScript type definitions.

Reply:
Not in the case where the ```intl``` props COULD came from ELSEWHERE, so implementing it as an INTERFACE instead...
@nyanofthemoon tried to explain this to you; you did hear it but seem to refuse to listen to her. 
This way, we have the flexibility for other team (in a different project) to choose a different  way of implementing internalization...

Reply:
@latch2112 ... Wow ... The argument we had yesterday at the office was about the removal of an **unused and bugged state** declaration **NOT** about this. You even raised your voice at me for having an opinion, remember? At the end we ended up removing it because it was non sense to keep it since it was bugged anyway.

You can keep defending your personal friend all you want. This will NOT change the fact that your app is currently full of bugs and used anti-patterns as described here: https://github.com/cr-ste-justine/clin-frontend/issues/80